subprojects/workers/src/main/java/org/gradle/workers/internal/IsolationMode.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal;
+
+/**
+ * Isolation mode for workers.
+ */
+public enum IsolationMode {
```

Comment:
Having a second (internal) copy of this class could be confusing.  I think we could just get rid of this class altogether now.  It only gets used by `AbstractDaemonCompiler` now to determine what sort of `WorkerFactory` it's been given.  Instead, I wonder if we could introduce a `DaemonCompilerWorkerFactory` (or some such) that wraps the original `WorkerFactory` instance and takes a T spec as its argument.  Then, `AbstractDaemonCompiler` just gets one of these and calls `getWorker(spec)` and it takes care of creating a `WorkerRequirement` and calling the underlying `WorkerFactory`.  We could inject it from the compiler factories.  Does that make sense?  Happy to chat about it if not.